src/components/SearchInput/index.jsx

```diff
@@ -1,19 +1,66 @@
-import { useRef, useState } from "react";
+import { useRef, useState, useEffect } from "react";
+import axios from "axios";
+import useUserInputStore from "../../store/store";
 
 function SearchInput() {
-  const [query, setQuery] = useState("");
+  const { userInput, setUserInput } = useUserInputStore();
+  const [autoCompletions, setAutoCompletions] = useState([]);
+  const [selectedItemIndex, setSelectedItemIndex] = useState(-1);
   const inputRef = useRef(null);
 
-  function handleQueryChange(event) {
-    setQuery(event.target.value);
+  function handleUserInputChange(event) {
+    setUserInput(event.target.value);
   }
 
-  function handleSubmit(event) {
-    event.preventDefault();
+  useEffect(() => {
+    async function getAutoCompletions(userInput) {
+      const baseURL = "http://localhost:3000";
+
+      try {
+        const response = await axios.get(`${baseURL  }/auto-completions`, {
+          params: { userInput },
+        });
+        const {data} = response;
+
+        setAutoCompletions(data);
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    getAutoCompletions(userInput);
+  }, [userInput]);
+
+  function handleArrowKeyPress(event) {
+    if (event.key === "ArrowDown") {
+      event.preventDefault();
+
+      setSelectedItemIndex((prevIndex) =>
+        prevIndex < autoCompletions.length - 1 ? prevIndex + 1 : 0,
+      );
+    } else if (event.key === "ArrowUp") {
+      event.preventDefault();
+
+      setSelectedItemIndex((prevIndex) =>
+        prevIndex > 0 ? prevIndex - 1 : autoCompletions.length - 1,
+      );
+    }
+  }
```

Comment:
```suggestion
  function handleArrowKeyPress(event) {
    event.preventDefault();
    
    if (event.key === "ArrowDown") {
      setSelectedItemIndex((prevIndex) =>
        prevIndex < autoCompletions.length - 1 ? prevIndex + 1 : 0,
      );
    } else if (event.key === "ArrowUp") {
      setSelectedItemIndex((prevIndex) =>
        prevIndex > 0 ? prevIndex - 1 : autoCompletions.length - 1,
      );
    }
  }
```
preventDefault를 맨위로 올리면 좋을 것 같아요